### PR TITLE
Development to master, sprint 6

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,12 +1,46 @@
...
+#include <stdint.h>
 
 int main() {
     WDT->WDT_MR = WDT_MR_WDDIS;
 
     hwlib::wait_ms(1000);
-    MovingPlatform::Platform test;
-    hwlib::cout << test.getWheelSize() << hwlib::endl;
 
+    UARTLib::HardwareUART serialCon(115200, UARTLib::UARTController::TWO);
```
Comment (elsavira) [inline src/main.cpp]: No reason to abbreviate variable name here. serialCon-what?

File: src/moving_platform.cpp [hunk 2]
```diff
@@ -2,27 +2,36 @@
 #include "wrap-hwlib.hpp"
 
 namespace MovingPlatform {
-// Platform(? & leftMotor, ? & rightMotor):
-// leftMotor(leftMotor), rightMotor(rightMotor)
-Platform::Platform() {
+Platform::Platform(UARTLib::UARTConnection &serialCon, hwlib::pin_in &hallSensorPinA, hwlib::pin_in &hallSensorPinB)
+    : serialCon(serialCon), motorController(Qik2S12V10(serialCon)), hallSensorPinA(hallSensorPinA), hallSensorPinB(hallSensorPinB) {
     wheelsize = 60;
```
Comment (elsavira) [inline src/moving_platform.cpp]: wheelSize according to the guidelines.

File: src/moving_platform.hpp [hunk 3]
```diff
@@ -17,18 +19,70 @@ namespace MovingPlatform {
...
-    //? & leftMotor;
-    //? & rightMotor;
+  private:
+    UARTLib::UARTConnection &serialCon;
+    Qik2S12V10 motorController;
+    hwlib::pin_in &hallSensorPinA;
+    hwlib::pin_in &hallSensorPinB;
+
     int16_t currentSpeedLeft, currentSpeedRight;
     int16_t wheelsize;
```
Comment (elsavira) [inline src/moving_platform.hpp]: As mentioned before; wheelSize

File: src/moving_platform.hpp [hunk 4]
```diff
@@ -17,18 +19,70 @@ namespace MovingPlatform {
...
-    Platform();
+  private:
+    /**
+     * @brief Moves the platform forward.
+     *
+     * Move the platform forward for *distanceMilliMeters*.
+     *
+     * @param[in] distanceMilliMeters The amount of millimeters the platform should move.
+     */
+    void movePlatformForward(int32_t distanceMilliMeters);
```
Comment (elsavira) [inline src/moving_platform.hpp]: Millimetres is 1 word, so: distanceMillimetres.

File: src/moving_platform.hpp [hunk 5]
```diff
@@ -17,18 +19,70 @@ namespace MovingPlatform {
...
+
+    /**
+     * @brief Moves the platform backward.
+     *
+     * Move the platform backward for *distanceMilliMeters*.
+     *
+     * @param[in] distanceMilliMeters The amount of millimeters the platform should move.
+     */
+    void movePlatformBackward(int32_t distanceMilliMeters);
```
Comment (elsavira) [inline src/moving_platform.hpp]: As mentioned above.

**Review by elsavira — CHANGES_REQUESTED** [review]
Some variable names need changing.

File: src/main.cpp (hunk 1 above)
Comment (SamZand) [reply src/main.cpp]: Changed to serialConnection

File: src/moving_platform.cpp (hunk 2 above)
Comment (SamZand) [reply src/moving_platform.cpp]: Done

File: src/moving_platform.hpp (hunk 3 above)
Comment (SamZand) [reply src/moving_platform.hpp]: Done

File: src/moving_platform.hpp (hunk 4 above)
Comment (SamZand) [reply src/moving_platform.hpp]: Changed to distanceMillimeters

File: src/moving_platform.hpp (hunk 5 above)
Comment (SamZand) [reply src/moving_platform.hpp]: Changed to distanceMillimeters

**Review by elsavira — APPROVED** [review]
I will condone the use of American English for this once. But I promise, it's Millimetres.

**Pull request merged**